File: back/src/main/java/solutiona/challenge/pickaboo/presentation/request/CreateUserRequestDto.java
package solutiona.challenge.pickaboo.presentation.request;

import java.time.LocalDate;
import java.util.List;

public record CreateUserRequestDto(
        String location,
        String language,
        LocalDate birth,
        String gender,
        String bloodType,
        String skinType,
        List<AllergyRequestDto> allergyList,
        List<OnboardingDiseaseRequestDto> onDiseaseList


) {
}

File: client/data/src/main/java/com/peekaboo/data/entity/request/auth/CreateUserRequestDto.kt
package com.peekaboo.data.entity.request.auth

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CreateUserRequestDto(
    @SerialName("location")
    val location: String,
    @SerialName("language")
    val language: String,
    @SerialName("birth")
    val birth: String,
    @SerialName("gender")
    val gender: String,
    @SerialName("bloodType")
    val bloodType: String,
    @SerialName("skinType")
    val skinType: String,
    @SerialName("allergyList")
    val allergyList: List<AllergyItem>,
    @SerialName("onDiseaseList")
    val onDiseaseList: List<DiseaseItem>,
) {
    @Serializable
    data class AllergyItem(
        @SerialName("description")
        val description: String,
    )

    @Serializable
    data class DiseaseItem(
        @SerialName("description")
        val description: String,
    )
}